liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/CreateServerMojo.java

```diff
@@ -174,4 +175,39 @@ private File getFileFromConfigDirectory(String file, File def) {
         } 
         return null;
     }
+    
+    /*
+     * return the filename of the project artifact to be installed by install-apps goal
+     */
+    private String getApplicationFilename() {
+        // project artifact has not be created yet when create-server goal is called in pre-package phase
+        String name = project.getBuild().getFinalName();
+        if (stripVersion) {
+            int versionBeginIndex = project.getBuild().getFinalName().lastIndexOf("-" + project.getVersion());
+            if ( versionBeginIndex != -1) {
+                name = project.getBuild().getFinalName().substring(0, versionBeginIndex);
+            }
+        }
+        
```

Comment:
finalName is default to ${artifactId}-${version} (see https://maven.apache.org/pom.html).  But it is often used in pom.xml to strip the version from the project output artifact filename.  Here is an example, https://github.com/WASdev/sample.servlet.jdbc/blob/master/pom.xml.  If you insist, I can change it to "name = project.getArtifactId();"